src/app.ts

```diff
@@ -0,0 +1,93 @@
+import express, { Application, Request, Response, NextFunction } from "express";
+import * as bodyParser from "body-parser";
+import { check, validationResult } from "express-validator";
+
+const app: Application = express();
+
+app.use(bodyParser.json());
+
+interface todo {
+  id: number;
+  text: string;
+  complete: boolean;
+}[];
+
+let todoRepository: todo[];
+
+todoRepository = [];
+
+/* post */
+
+app.post(
+  "/",
+  [check("text").isString(), check("complete").isBoolean()],
+  (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    const todo = {
+      id: todoRepository.length + 1,
```

Comment:
Don't use `length` for `id` because it's will be a problem when you delete an element and the insert new element. the old id will be replaced

Reply:
fixed on 5e1eb54c01ef4a6b08ebc2d1c553572e5fa17e1a